assets/src/main/java/bisq/asset/coins/Veil.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * This file is part of Bisq.
+ *
+ * Bisq is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.asset.coins;
+
+import bisq.asset.*;
+import org.bitcoinj.core.Address;
+import org.bitcoinj.core.AddressFormatException;
+
+public class Veil extends Coin {
+
+    public Veil() {
+        super("Veil", "VEIL", new VeilAddressValidator());
+    }
+
+    public static class VeilAddressValidator extends Base58BitcoinAddressValidator {
+
+        public VeilAddressValidator() {
+            super(new VeilParams());
+        }
+
+        @Override
+        public AddressValidationResult validate(String address) {
+            if (address.startsWith("V")) {
+                try {
+                    // Base58
+                    Address.fromBase58(getNetworkParameters(), address);
```

Comment:
Do not expose network parameters publicly. Just call `super.validate(address)`.

Reply:
Hi ok, no problem. 

Them are not exposed publicly, just made them protected to be able to get them in the future for bech32 and stealth addresses validation in case of not have them fully supported on the platform it selves.